### PR TITLE
Support Issues and Custom Fields

### PR DESCRIPTION
**Pull request by Etroid**
- Adds the ability to add custom fields to any given query.
- Stores the query params in the header when passing them along to our proxy endpoint.
- Supports both basic authentication (username and password) as well as API key.
- Adds the ability to query issues for a given project ID

### CODE REVIEW

**Comment by iamEAP** [comment]
It's Alive

File: src/main.js [hunk 1]
```diff
@@ -94,18 +94,37 @@ module.exports = function($, tableau, wdcw) {
...
-      if (wdcFields.coreFields.hasOwnProperty(column)) {
-        processedColumns.push({
-          name: column,
-          type: wdcFields.coreFields[column]
-        })
+        column,
+        customFields = this.getConnectionData().customFields,
+        objType = this.getConnectionData().objType;
+
+    tableau.log(objType);
```
Comment (iamEAP) [inline src/main.js]: - [ ] Stray log call.

File: src/main.js [hunk 2]
```diff
@@ -169,27 +191,31 @@ module.exports = function($, tableau, wdcw) {
...
-    // If opts.limit was passed, add a limit flag to the URL.
-    if (opts.limit) {
-      path += '&limit=' + opts.limit;
+    // If we have custom fields, append them to our query.
+    if (customFields.length) {
+      fields = fields.concat(customFields).join(',');
     }
 
-    return path;
+    console.log(data.projectID);
```
Comment (iamEAP) [inline src/main.js]: - [ ] 'Nother one.

File: src/wrapper.js [hunk 3]
```diff
@@ -268,19 +267,44 @@ var wdcw = window.wdcw || {};
...
 
       // Format connection data according to assumptions.
       $fields.map(function getValuesFromFields() {
-        var $this = $(this);
+        var $this = $(this),
             name = $this.attr('name');
-        if (name) {
-          data[name] = $this.val();
+
+        switch (name) {
```
Comment (iamEAP) [inline src/wrapper.js]: - [ ] I've had my eye on moving all of the logic in `wrapper.js` into its own installable/require-able package.  Is it possible to do this logic elsewhere in a way that keeps `wrapper.js` relatively generic?

**Comment by iamEAP** [comment]
This is awesome!  Was able to get this working in the simulator _and_ in Desktop! :tada: :tada: :tada:

File: src/wrapper.js (hunk 3 above)
Comment (Etroid) [inline src/wrapper.js]: Good call. I'll turn it into a helper function in the ```main.js``` instead.

**Comment by Etroid** [comment]
It's a very rudimentary __first__ approach to integrating multiple object types and custom fields, but I needed something quickly to get Daniel started with his dashboard. Happy to discuss future enhancements tomorrow.

**Comment by iamEAP** [comment]
:+1:

**Pull request merged**